### PR TITLE
chore(nodebuilder/tests): Clean up integration tests in `p2p_test`

### PR DESCRIPTION
**Pull request by renaynay**
Cleans up swamp tests in `p2p_test.go`.

Based on #2162

### CODE REVIEW

File: nodebuilder/tests/sync_test.go [hunk 1]
```diff
@@ -265,54 +256,52 @@ Steps:
...
 	err = light.Start(ctx)
 	require.NoError(t, err)
-
-	h, err = light.HeaderServ.GetByHeight(ctx, 50)
+	err = light.HeaderServ.SyncWait(ctx)
 	require.NoError(t, err)
 
-	assert.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, 50))
+	// wait for the core block filling process to exit
+	require.NoError(t, <-fillDn)
```
Comment (Wondertan) [inline nodebuilder/tests/sync_test.go:306]: we don't trust that chans will get closed

File: nodebuilder/tests/sync_test.go [hunk 2]
```diff
@@ -265,54 +256,52 @@ Steps:
...
-	addrs, err = peer.AddrInfoToP2pAddrs(host.InfoFromHost(full.Host))
+	err = full.HeaderServ.SyncWait(ctx)
 	require.NoError(t, err)
 
+	// create a LN with both FN and BN as trusted peers
 	cfg = nodebuilder.DefaultConfig(node.Light)
-	cfg.Header.TrustedPeers = append(cfg.Header.TrustedPeers, addrs[0].String())
+	fullInfo, err := peer.AddrInfoToP2pAddrs(host.InfoFromHost(full.Host))
+	require.NoError(t, err)
+	cfg.Header.TrustedPeers = append(cfg.Header.TrustedPeers, bridgeInfo[0].String(), fullInfo[0].String())
```
Comment (Wondertan) [inline nodebuilder/tests/sync_test.go:296]: I think we should hide this as part of the node setup. IIRC, I did that.

File: nodebuilder/tests/sync_test.go [hunk 3]
```diff
@@ -265,54 +256,52 @@ Steps:
...
 5. Start a FN
-6. Check FN is synced to height 30
-7. Create a Light Node(LN) with a connection to BN, FN as trusted peers
+6. Check FN is synced to network head
+7. Create a Light Node(LN) with a connection to BN and FN as trusted peers
 8. Start LN
-9. Check LN is synced to height 50
+9. Check LN is synced to network head.
 */
 func TestSyncLightWithTrustedPeers(t *testing.T) {
```
Comment (Wondertan) [inline nodebuilder/tests/sync_test.go:264]: It's now the same as TestSyncLightWithTrustedPeers from what it tests perspective or do I miss something?

File: nodebuilder/tests/sync_test.go (hunk 1 above)
Comment (Wondertan) [reply nodebuilder/tests/sync_test.go:306]: Relates to all places

File: nodebuilder/tests/sync_test.go [hunk 4]
```diff
@@ -17,60 +17,99 @@ import (
...
-	assert.NoError(t, err)
-
-	err = light.DASer.WaitCatchUp(ctx)
+	h, err := bridge.HeaderServ.GetByHeight(ctx, numBlocks)
 	require.NoError(t, err)
-
-	assert.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, 30))
+	require.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, numBlocks))
+
+	t.Run("light sync against bridge", func(t *testing.T) {
```
Comment (Wondertan) [inline nodebuilder/tests/sync_test.go:64]: Nice

File: nodebuilder/tests/sync_test.go [hunk 5]
```diff
@@ -17,60 +17,99 @@ import (
...
+		// create a light node that is connected to the bridge node as
+		// a bootstrapper
+		cfg := nodebuilder.DefaultConfig(node.Light)
+		addrs, err := peer.AddrInfoToP2pAddrs(host.InfoFromHost(bridge.Host))
+		require.NoError(t, err)
+		cfg.Header.TrustedPeers = append(cfg.Header.TrustedPeers, addrs[0].String())
+		light := sw.NewNodeWithConfig(node.Light, cfg)
```
Comment (Wondertan) [inline nodebuilder/tests/sync_test.go:71]: This really asks to be hidden away

File: nodebuilder/tests/sync_test.go [hunk 6]
```diff
@@ -17,60 +17,99 @@ import (
...
-	bsize  = 16
-	btime  = time.Millisecond * 300
+	numBlocks = 20
+	bsize     = 16
+	btime     = time.Millisecond * 300
 )
 
 /*
-Test-Case: Sync a Light Node with a Bridge Node(includes DASing of non-empty blocks)
+Test-Case: Header and block/sample sync against a Bridge Node of non-empty blocks.
```
Comment (Wondertan) [inline nodebuilder/tests/sync_test.go:26]: I would keep some test with empty blocks as well

**Review by Wondertan — COMMENTED** [review]
Can we remove unrelated tests from the diff?

File: nodebuilder/tests/sync_test.go (hunk 1 above)
Comment (renaynay) [reply nodebuilder/tests/sync_test.go:306]: will wrap w select on ctx

Comment (renaynay) [reply nodebuilder/tests/sync_test.go]: https://github.com/celestiaorg/celestia-node/pull/2162/commits/0c19eb09514b56d1dad13d4ac251270cf9c2f2a1

File: nodebuilder/tests/sync_test.go (hunk 2 above)
Comment (renaynay) [reply nodebuilder/tests/sync_test.go]: https://github.com/celestiaorg/celestia-node/pull/2162/commits/0c19eb09514b56d1dad13d4ac251270cf9c2f2a1

File: nodebuilder/tests/sync_test.go (hunk 5 above)
Comment (renaynay) [reply nodebuilder/tests/sync_test.go]: https://github.com/celestiaorg/celestia-node/pull/2162/commits/0c19eb09514b56d1dad13d4ac251270cf9c2f2a1

File: nodebuilder/tests/sync_test.go (hunk 6 above)
Comment (renaynay) [reply nodebuilder/tests/sync_test.go:24]: https://github.com/celestiaorg/celestia-node/pull/2162/commits/09cb4987429a5d49c9807ac9eda8a4f226945b17

**Review by distractedm1nd — APPROVED** [review]

File: nodebuilder/tests/p2p_test.go [hunk 7]
```diff
@@ -61,19 +62,27 @@ Steps:
 8. Check LN is not allowed to dial with FN
 */
 func TestAddPeerToBlackList(t *testing.T) {
```
Comment (walldiss) [inline nodebuilder/tests/p2p_test.go:64]: Weird test, do you know what is the reason to have it in swamp? Looks like a unit test for connGater, which is external dependancy, so maybe it should be removed.

File: nodebuilder/tests/p2p_test.go [hunk 8]
```diff
@@ -86,131 +95,123 @@ Steps:
...
-			}
+
+	for {
+		if ctx.Err() != nil {
+			t.Fatal(ctx.Err())
+		}
+		if light.Host.Network().Connectedness(host.InfoFromHost(full.Host).ID) == network.Connected {
+			// LN discovered FN successfully and is now connected
+			break
 		}
```
Comment (walldiss) [inline nodebuilder/tests/p2p_test.go:150]: Polling loop without delay would eat cpu in tests and slow down CI. Why not stick with event bus as it used to be?

File: nodebuilder/tests/p2p_test.go [hunk 9]
```diff
@@ -86,131 +95,123 @@ Steps:
...
 1. Create a Bridge Node(BN)
 2. Start a BN
 3. Create 2 full nodes with bridge node as bootstrapper peer and start them
 4. Check that nodes are connected to each other
-5. Create one more node with disabled discovery
-6. Disconnect FNs from each other
-7. Check that the last FN is connected to one of the nodes
+5. Disconnect the FNs
+6. Create one more node with disabled discovery
+7. Check that the FN with discovery disabled is still found by the other two FNs
```
Comment (walldiss) [inline nodebuilder/tests/p2p_test.go:163]: I wonder, if test can skip steps 4,5: connecting FN and then disconnecting them right after. Unless theres is the reason to have those?

File: nodebuilder/tests/p2p_test.go [hunk 10]
```diff
@@ -86,131 +95,123 @@ Steps:
...
 Steps:
 1. Create a Bridge Node(BN)
 2. Start a BN
 3. Create 2 full nodes with bridge node as bootstrapper peer and start them
 4. Check that nodes are connected to each other
-5. Create one more node with disabled discovery
-6. Disconnect FNs from each other
-7. Check that the last FN is connected to one of the nodes
+5. Disconnect the FNs
+6. Create one more node with disabled discovery
```
Comment (walldiss) [inline nodebuilder/tests/p2p_test.go:162]: After reading description I was a bit confused how node could be discovered, when it has discovery disabled. Turned out it description is a bit misleading. According to code last node is not really have disabled discovery, but rather it is not searching for other nodes, but still advertises itself to allow it to be discovered by others.

File: nodebuilder/tests/sync_test.go [hunk 11]
```diff
@@ -17,61 +15,184 @@ import (
...
-
 	err := bridge.Start(ctx)
 	require.NoError(t, err)
 
-	h, err := bridge.HeaderServ.WaitForHeight(ctx, 20)
+	h, err := bridge.HeaderServ.WaitForHeight(ctx, numBlocks)
 	require.NoError(t, err)
+	require.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockHashByHeight(ctx, numBlocks))
+
+	t.Run("light sync against bridge", func(t *testing.T) {
```
Comment (walldiss) [inline nodebuilder/tests/sync_test.go:63]: Suggestion: Looks like at this stage, blocks and headers are already produced by BN and there the Light and Full node only need to sync. Maybe we can use lower context, that is local for each `t.Run` to allow faster end if tests fails.

**Review by walldiss — COMMENTED** [review]
Thank you for cleanup! Swamp looks so much better!

**Comment by Wondertan** [comment]
@walldiss, @distractedm1nd, as this PR is based on https://github.com/celestiaorg/celestia-node/pull/2162
let's review it first, unless I miss something @renaynay

File: nodebuilder/tests/p2p_test.go (hunk 7 above)
Comment (renaynay) [reply nodebuilder/tests/p2p_test.go:64]: Yeah I'm also unsure as to why that test is there. I'm happy to remove it - cc @Wondertan

Comment (Wondertan) [reply nodebuilder/tests/p2p_test.go]: I am also happy to remove it. IIRC, it was added by @Bidon15 to ensure blacklisting works correctly once we exposed he feature for testground tests

File: nodebuilder/tests/sync_test.go (hunk 11 above)
Comment (renaynay) [reply nodebuilder/tests/sync_test.go:63]: Hmm why lower the context? We still need to wait for DASing too.

File: nodebuilder/tests/p2p_test.go (hunk 8 above)
Comment (renaynay) [reply nodebuilder/tests/p2p_test.go:115]: listening for a connected event does not work for some reason even though it definitely should. Will continue investigating.

Comment (walldiss) [reply nodebuilder/tests/p2p_test.go:115]: Probably because you need to spin out listener before even happens. With for loop you check after the event

Comment (renaynay) [reply nodebuilder/tests/p2p_test.go:115]: I started an event listening loop before the event happens and it still doesn't work. Will continue investigating but if this is blocking, we can move on other cleanussy PRs.

**Review by distractedm1nd — APPROVED** [review]

File: nodebuilder/tests/p2p_test.go (hunk 8 above)
Comment (renaynay) [reply nodebuilder/tests/p2p_test.go:115]: Investigate why subscription isn't working in this test.

Comment (renaynay) [reply nodebuilder/tests/p2p_test.go:115]: https://github.com/celestiaorg/celestia-node/issues/2649

File: nodebuilder/tests/p2p_test.go (hunk 10 above)
Comment (renaynay) [reply nodebuilder/tests/p2p_test.go]: added description there

File: nodebuilder/tests/p2p_test.go (hunk 9 above)
Comment (renaynay) [reply nodebuilder/tests/p2p_test.go]: added description there - no reason, just additional check that they're able to discover each other

**Review by walldiss — APPROVED** [review]

**Review by distractedm1nd — APPROVED** [review]

**Pull request merged**